packages/mapviewer/src/api/layers/WMSCapabilitiesParser.class.js

```diff
@@ -433,6 +433,25 @@ export default class WMSCapabilitiesParser {
 
     _getLayerLegends(layerId, layer) {
         const styles = layer.Style?.filter((s) => s.LegendURL?.length > 0) ?? []
+        // if we do not have access to the legend in pure WMS fashion, we check if this
+        // WMS follows the SLD specification, and if we can get it from there.
+        if (
+            styles.length === 0 &&
+            layer.queryable &&
+            !!this.Capability.UserDefinedSymbolization?.SupportSLD
+        ) {
+            const getLegendGraphicBaseUrl =
+                this.Capability.Request.GetLegendGraphic?.DCPType[0]?.HTTP?.Get?.OnlineResource
+            const getLegendGraphicFormat = this.Capability.Request.GetLegendGraphic?.Format[0]
+            if (!!getLegendGraphicBaseUrl && !!getLegendGraphicFormat) {
+                return [
+                    {
+                        url: `${getLegendGraphicBaseUrl}SERVICE=WMS&REQUEST=GetLegendGraphic&ERSION=${this.version}&FORMAT=${getLegendGraphicFormat}&LAYER=${layerId}&SLD_VERSION=1.1.0`,
```

Comment:
Typo here with `VERSION` param
That's quite a lot of params, might be better to describe them separately as an object, such as

```Javascript
const urlParams = {
    SERVICE: 'WMS',
    REQUEST: 'GetLegendGraphic',
    VERSION: this.version,
    // etc ...
}
```

And then create the URL by doing something like

```Javascript
Object.entries(urlParams).reduce((acc, [key, value], index) => {
  const prefix = index === 0 ? "?" : "&";
  return acc + `${prefix}${key}=${value}`;
}, "");
```

or use the URLSearchParams API

(I'm not even sure you need the part that checks for the `?` instead of `&`, as a WMS base URL should always end with `?` AFAIK)

Reply:
doing a URLSearchParams object and using it to generate the URL would be better IMO. We might still want to add a quick check for the `?` at the end of the base url, as I think you're correct in saying a WMS base URL should always end with it, but I would prefer to be on the safe side just in case there is one implementation where it was forgotten.